### PR TITLE
SUS-4765: check opt-out status for Qualaroo, GA, Comscore and Quantcast

### PR DESCRIPTION
**Pull request by mszabo-wikia**
Check opt-out status for Qualaroo, GA, Comscore and Quantcast before loading external code and remove all tracking pixels that'd have been loaded if JavaScript was disabled.

https://wikia-inc.atlassian.net/browse/SUS-4765

### CODE REVIEW

File: extensions/wikia/AnalyticsEngine/AnalyticsProviderComscore.php [hunk 1]
```diff
@@ -16,22 +16,22 @@ function trackEvent($event, $eventDetails=array()){
...
-	_comscore.push({ c1: "2", c2: "'.static::$PARTNER_ID.'",
-	options: {
-		url_append: "'.static::$COMSCORE_KEYWORD_KEYNAME.'='.static::getC7Value().'"
-	}
+require(["wikia.trackingOptOut"], function (trackingOptOut) {
+	trackingOptOut.ifNotOptedOut(function () {
+		window._comscore = window._comscore || [];
+		window._comscore.push({ c1: "2", c2: "' . static::$PARTNER_ID . '",
+			options: {
+				url_append: "' . static::$COMSCORE_KEYWORD_KEYNAME . '=' . static::getC7Value() . '"
```
Comment (nandy-andy) [inline extensions/wikia/AnalyticsEngine/AnalyticsProviderComscore.php:24]: Don't we have to `urlencode()` the whole `url_append` value?

File: resources/wikia/modules/spec/trackingOptOut.spec.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
+describe('Tracking opt-out', function () {
```
Comment (nandy-andy) [inline resources/wikia/modules/spec/trackingOptOut.spec.js:1]: 💎

**Review by nandy-andy — COMMENTED** [review]
Just one question ;-) Thanks for unit tests! ❤️

File: extensions/wikia/AnalyticsEngine/AnalyticsProviderComscore.php (hunk 1 above)
Comment (mszabo-wikia) [reply extensions/wikia/AnalyticsEngine/AnalyticsProviderComscore.php:24]: I just copied what's been already there for years 😉 If we find it's broken perhaps we should sunset this feature as unused. 🙈

**Pull request merged**